### DISCUSSION TITLE
Migration to independent platform for provision of source code

### DISCUSSION THREAD

**Discussion by linux-lukas** [solokeys/solo2#121] [General]
### A. Problem / Goal 

Since the purchase of GitHub by Microsoft in 2018, a dependence on the BigTech corporation can no longer be denied. 

On the one hand, I can understand why GitHub was chosen as the platform for making source code available: "Everyone is here".

On the other hand, I see the danger of a vendor-lockin effect and that open source projects become centrally dependent on Microsoft. In my eyes, this is very dangerous for free and open source software and hardware projects.

In the medium and long term, the goal would be to become independent of GitHub and thus of Microsoft. The Gitea-based Codeberg project of Codeberg e. V. in Berlin would be a good choice here.

There are also (legal) problems with compliance with the licence of GitHub functions, such as the co-pilot. 

###  B. Solution 

My considered solution to the problem described in A. would be the following:

1. A user of this open source project creates a user account on https://codeberg.org/
2. If necessary: This user creates an organisation for the project.
3. A "personal access token" is created on the GitHub account, which has appropriate rights to the organisation repositories, using the developer options in the settings.
4. all repositories would be migrated with this access token into the ownership of the organisation created in step two.

Regarding step four, there is an entry in the documentation of Codeberg: https://docs.codeberg.org/advanced/migrating-repos/ 

### C. Alternatives

A possible alternative would be to perform the first three steps as described in B. A possible alternative would be to perform the first three steps as described in B., and modify the fourth step to include a mirror of GitHub. So that all issues and such that would be created in the GitHub repository would be transferred to the Codeberg repository.

### D. Responsibilities

I would see the responsibility in the owners of the repository and, if necessary, additional project participants.

###  E. Other

Basically, a look at the documentation of Codeberg is not unwise: https://docs.codeberg.org 

Should it be necessary to manage repositories in organisations, this is also possible under Codeberg, see: https://docs.codeberg.org/collaborating/create-organization/ 

Regarding licensing there is a page in the documentation of Codeberg: 
https://docs.codeberg.org/getting-started/licensing/ 

It would gradually affect all repositories in the GitHub organisation.

The Terms of Use of Codeberg can be viewed at: https://codeberg.org/Codeberg/org/src/branch/main/TermsOfUse.md

### F. Risk

Last but not least, it must be assumed that people could potentially create fewer issues because it is a new platform and it is less known. It remains to be seen how and when the principle of decentralisation or federation will be implemented in Gitea, on which Codeberg, the GitHub alternative, is based, see the following article: https://social.exozy.me/@ta180m/108631221939677386

**Comment by needs-coffee** [comment]
From reading your post it does not appear that this is being made in good faith, especially considering your GitHub profile seems to predominantly used to spam this post over issues and discussions of many repositories. 

**Are you affiliated with Codeberg?** 
Your enthusiasm for a very specific GitHub alternative is suspicious in this regard and if you are affiliated, you should be stating this explicitly. 

Regardless if whether your are affiliated with Codeberg or not - you are clearly new here as this project has many, many more issues that are more important than where git repository has been hosted.

**Reply by linux-lukas** [reply]
With my contribution I wanted to draw attention to the dependence on large US technology companies, which I consider enormously dangerous for Open Source Projects (OSP) and I consider independence important for OSP. I was and am aware of the fact that there are certainly more priority issues. 

If this did not come across as such, I am sorry and would like to apologise here.

No, I am not partnered with Codeberg. I only became aware of the Codeberg project some time ago as an independent European alternative to GitHub and over time I found it to be a good alternative and wanted to draw attention to the fact that there is more than just GitHub. I personally use it even more than GitHub.
